sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -831,24 +839,17 @@ private static ArraySegment<byte> StreamToBytes(Stream stream)
             }
             else
             {
-                using (var memoryStream = new MemoryStream(512))
+                using var memoryStream = new MemoryStream(StreamBufferSizeInBytes);
```

Comment:
If we can get away with extracting the buffer without a copy for a `MemoryStream`, should we consider a switch here and only allocate a new one if it's not already a memory stream?